packages/eas-cli/src/build/build.ts

```diff
@@ -649,3 +654,63 @@ function formatAccountSubscriptionsUrl(accountName: string): string {
     getExpoWebsiteBaseUrl()
   ).toString();
 }
+
+async function createAndUploadFingerprintAsync<T extends Platform>(
+  ctx: BuildContext<T>
+): Promise<{
+  runtimeVersion?: string;
+  fingerprintGCSBucketKey?: string;
+}> {
+  const resolvedRuntimeVersion = await resolveRuntimeVersionAsync({
+    exp: ctx.exp,
+    platform: ctx.platform,
+    workflow: ctx.workflow,
+    projectDir: ctx.projectDir,
+    env: ctx.buildProfile.env,
+    cwd: ctx.projectDir,
+  });
+
+  /**
+   * It's ok for fingerprintSources to be empty
+   * fingerprintSources only exist if the project is using runtimeVersion.policy: fingerprint
+   */
+  if (!resolvedRuntimeVersion?.fingerprintSources) {
+    return {
+      runtimeVersion: resolvedRuntimeVersion?.runtimeVersion ?? undefined,
+    };
+  }
+
+  const fingerprintLocation = path.join(getTmpDirectory(), `${uuidv4()}-runtime-fingerprint.json`);
```

Comment:
Hm. How should I do that? It doesn't seem like the [other places `getTmpDirectory` is used](https://github.com/search?q=repo:expo/eas-cli%20getTmpDirectory&type=code) do an additional check. Should this be added everywhere?

Reply:
Can you explain what you mean by checking that it exists? That the path is non-empty or that the folder exists? I copied this implementation from [metadata](https://github.com/expo/eas-cli/blob/1c61465c5f7526770d3fc489828bef18803dd365/packages/eas-cli/src/build/utils/repository.ts#L103C9-L103C25) btw

Reply:
>That the path is non-empty or that the folder exists? 

I want us to check if the folder already exists.

>I copied this implementation from [metadata](https://github.com/expo/eas-cli/blob/1c61465c5f7526770d3fc489828bef18803dd365/packages/eas-cli/src/build/utils/repository.ts#L103C9-L103C25) btw

Oh okay, I believe it works for metadata, and here without this check because it is already created when uploading project archive tarball (`fs.mkdirp`)
https://github.com/expo/eas-cli/blob/0516b507bce06c136e38700e0d333675e719a8ba/packages/eas-cli/src/build/utils/repository.ts#L122-L126

I believe we can add the `fs.mkdirp` call here for extra safety so that refactor in the project tarball upload code won't break anything here. We can also fix it in metadata upload.

Do you think it makes sense? What do you think?

Reply:
Makes sense to me! Added here for both metadata and fingerprint: https://github.com/expo/eas-cli/pull/2422/commits/005c0388c5b039b9dbbca4f96741c207c99dda15